File: JavaScript/DataStructures/stack.js
function Stack() {
    this.stack = []
}

Stack.prototype.push = function(el) {
    this.stack.push(el);
    return el;
}

Stack.prototype.pop = function() {
    this.stack.pop();
}

Stack.prototype.peek = function() {
    return this.stack[this.stack.length - 1];
}

let bass = new Stack();
console.log(bass.push("bass"));
console.log(bass.peek());

File: RUBY/DataStructures/stack.rb
class Stack
    def initialize
        @stack = []
    end

    def push(el)
        stack.push(el)
        el
    end

    def pop
        stack.pop
    end

    def peek
        stack.last
    end

    private
    
    attr_reader :stack
end

rover = Stack.new
print rover.push("wait")
print rover.peek